uefi/src/mem/aligned_buffer.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use alloc::alloc::{alloc, dealloc, Layout, LayoutError};
+use core::error::Error;
+use core::fmt;
+
+/// Helper class to maintain the lifetime of a memory region allocated with a non-standard alignment.
+/// Facilitates RAII to properly deallocate when lifetime of the object ends.
+///
+/// Note: This uses the global Rust allocator under the hood.
+#[derive(Debug)]
+pub struct AlignedBuffer {
+    ptr: *mut u8,
```

Comment:
Do we want to use

```suggestion
    ptr: NonNull<u8>,
```

here instead? @nicholasbishop  what do you think?